src/MusicStore/Components/AnnouncementComponent.cs

```diff
@@ -10,18 +10,23 @@ namespace MusicStore.Components
     [ViewComponent(Name = "Announcement")]
     public class AnnouncementComponent : ViewComponent
     {
-        private readonly MusicStoreContext _dbContext;
-        private readonly IMemoryCache _cache;
+        [Activate]
+        public MusicStoreContext DbContext
+        {
+            get;
+            set;
+        }
 
-        public AnnouncementComponent(MusicStoreContext dbContext, IMemoryCache memoryCache)
+        [Activate]
+        public IMemoryCache MemoryCache
```

Comment:
Why moving these to activates properties makes things better?

-----Original Message-----
From: "Youngjune Hong" <notifications@github.com>
Sent: ‎2/‎9/‎2015 8:08 PM
To: "aspnet/MusicStore" <MusicStore@noreply.github.com>
Cc: "Yishai Galatzer" <yishai_galatzer@yahoo.com>
Subject: Re: [MusicStore] Updating MusicStore for better unit-testability(#382)

In src/MusicStore/Components/AnnouncementComponent.cs:
>  
> -        public AnnouncementComponent(MusicStoreContext dbContext, IMemoryCache memoryCache)
> +        [Activate]
> +        public IMemoryCache MemoryCache
@Praburaj Well. maybe we can use Cache just like using dbContext for MusicStoreContext?? I can change them to Cache if @pranavkm agree. 
—
Reply to this email directly or view it on GitHub.